src/shims/unix/foreign_items.rs

```diff
@@ -138,6 +138,11 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 let result = this.readlink(pathname, buf, bufsize)?;
                 this.write_scalar(Scalar::from_machine_isize(result, this), dest)?;
             }
+            "realpath" | "realpath$DARWIN_EXTSN" => {
```

Comment:
For other shims if macOS has a strange link name for it we usually put it in the `macos/foreign_items` file.  So probably it'd be better to also do this here.